roboteam_ai/src/control/ballHandling/BallHandlePosControl.cpp

```diff
@@ -250,17 +269,17 @@ RobotCommand BallHandlePosControl::goToMovingBall() {
     Vector2 projectionPosition = ballLine.project(robot->pos);
     double robotToProjectionDistance = (projectionPosition - robot->pos).length();
     double ballToProjectionDistance = (projectionPosition - ball->pos).length();
-    const double averageRobotVelocity = 1.5; // ms-1
+    const double averageRobotInterceptVelocity = 1.5; // ms-1
 
     Angle robotAngleTowardsBallVel = (robot->pos - ball->pos).toAngle() - ball->vel.toAngle();
     bool robotIsBehindBall = fabs(robotAngleTowardsBallVel) < M_PI_4;
 
-    bool robotCanInterceptBall = robotToProjectionDistance/averageRobotVelocity <
+    bool robotCanInterceptBall = robotToProjectionDistance/averageRobotInterceptVelocity <
             ballToProjectionDistance/ball->vel.length();
 
     if (robotIsBehindBall && robotCanInterceptBall) {
         numTreesTarget = projectionPosition;
-        auto robotCommand = NumTreePosControl::getRobotCommand(robot, numTreesTarget);
+        auto robotCommand = BasicPosControl::getRobotCommand(robot, numTreesTarget);
```

Comment:
Take the posController of the robot

Reply:
this is indirectly the poscontroller of the robot.
I'm calling the overwritten function of this

Reply:
that is so that we have the same variables for 'avoid defense area' and 'avoid ball' etc etc

Reply:
the function call is not a static function call, although it might look like it is.
 in fact, you can't even call non static functions in the first place.